File: sample-android/src/main/kotlin/dev/bmcreations/phantom/connect/sample/components/WalletInfoSection.kt
package dev.bmcreations.phantom.connect.sample.components

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.text.selection.SelectionContainer
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import dev.bmcreations.phantom.connect.PhantomSession
import dev.bmcreations.phantom.connect.sample.theme.PhantomPurple

@Composable
internal fun WalletInfoSection(session: PhantomSession) {
    Text(
        text = "Wallet ID:",
        style = MaterialTheme.typography.bodySmall,
        fontWeight = FontWeight.Bold,
    )
    SelectionContainer {
        Text(
            text = session.walletId,
            style = MaterialTheme.typography.bodySmall,
            fontFamily = FontFamily.Monospace,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
        )
    }
    if (session.addresses.isNotEmpty()) {
        Spacer(Modifier.height(4.dp))
        Text(
            text = "Addresses:",
            style = MaterialTheme.typography.bodySmall,
            fontWeight = FontWeight.Bold,
        )
        session.addresses.forEach { addr ->
            Column(modifier = Modifier.padding(start = 8.dp, top = 4.dp)) {
                Text(
                    text = "${addr.chain.id.uppercase()}:",
                    style = MaterialTheme.typography.bodySmall,
                    fontWeight = FontWeight.Medium,
                    color = PhantomPurple,
                )
                SelectionContainer {
                    Text(
                        text = addr.address,
                        style = MaterialTheme.typography.bodySmall,
                        fontFamily = FontFamily.Monospace,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis,
                    )
                }
            }
        }
    }
}

File: sample-ios/PhantomSample/Components/WalletInfoSection.swift
import SwiftUI
import PhantomConnectSDK

struct WalletInfoSection: View {
    let session: PhantomWalletSession

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text("Wallet ID:")
                .font(.caption)
                .fontWeight(.bold)
            Text(session.walletId)
                .font(.caption)
                .monospaced()
                .lineLimit(1)
                .truncationMode(.middle)
                .textSelection(.enabled)

            if !session.addresses.isEmpty {
                Spacer().frame(height: 4)
                Text("Addresses:")
                    .font(.caption)
                    .fontWeight(.bold)

                ForEach(session.addresses, id: \.address) { addr in
                    VStack(alignment: .leading, spacing: 2) {
                        Text("\(addr.chain.uppercased()):")
                            .font(.caption)
                            .fontWeight(.medium)
                            .foregroundColor(PhantomColors.purple)
                        Text(addr.address)
                            .font(.caption)
                            .monospaced()
                            .lineLimit(1)
                            .truncationMode(.middle)
                            .textSelection(.enabled)
                    }
                    .padding(.leading, 8)
                    .padding(.top, 4)
                }
            }
        }
    }
}
